packages/template-retail-react-app/app/pages/product-list/index.jsx

```diff
@@ -397,6 +399,15 @@ const ProductList = (props) => {
         }
     }, [productSearchResult])
 
+    /**************** Shopper Context ****************/
+    const {isUpdating: isUpdatingShopperContext} = useUpdateShopperContext()
+
+    useEffect(() => {
+        if (!isUpdatingShopperContext) {
+            refetch()
+        }
+    }, [isUpdatingShopperContext])
```

Comment:
Here we now refetch the `useProductSearch` hook everytime the shopper context update completes.

Pro:
- control exactly what is refetched from each component

Con:
- each component that uses shopper context will need to call the `useUpdateShopperContext()` and have a `useEffect` to refetch the data that needs to be refetched

Reply:
Why are we putting the context in only plp? What happens if a user lands on PDP after shopper context has been applies?